packages/firebase_data_connect/firebase_data_connect/lib/src/common/dataconnect_error.dart

```diff
@@ -28,6 +28,44 @@ class DataConnectError extends FirebaseException {
   final DataConnectErrorCode dataConnectErrorCode;
 }
 
+/// Error thrown when an operation is partially successful.
+class DataConnectOperationError<T> extends DataConnectError {
+  DataConnectOperationError(
+      DataConnectErrorCode code, String message, this.response)
+      : super(code, message);
+  final DataConnectOperationFailureResponse<T> response;
+}
+
+/// Nested class containing errors and decoded data.
+class DataConnectOperationFailureResponse<T> {
+  DataConnectOperationFailureResponse(this.errors, this.data, this.decodedData);
+  final Map<String, dynamic>? data;
+  final List<DataConnectOperationFailureResponseErrorInfo> errors;
+  final T? decodedData;
+}
+
+/// Error information per error.
+class DataConnectOperationFailureResponseErrorInfo {
+  DataConnectOperationFailureResponseErrorInfo(this.path, this.message);
+  String message;
+  List<DataConnectOperationFailureErrorInfoPathSegment> path;
+}
+
+/// Path where error occurred.
+sealed class DataConnectOperationFailureErrorInfoPathSegment {}
```

Comment:
Can we name this as just DataConnectPathSegment? It will have reusabality outside of an error context too.

Reply:
Done.